packages/http-server-csharp/src/lib/service.ts

```diff
@@ -268,30 +272,120 @@ export async function $onEmit(context: EmitContext<CSharpServiceEmitterOptions>)
         parts.forEach((p) => this.emitter.emitType(p));
         return "";
       }
-      const className = ensureCSharpIdentifier(this.emitter.getProgram(), model, name);
+      const isErrorType = isErrorModel(this.emitter.getProgram(), model);
+      const baseModelRef = model.baseModel
+        ? code`: ${this.emitter.emitTypeReference(model.baseModel)}`
+        : "";
+      const baseClass = baseModelRef || (isErrorType ? ": HttpServiceException" : "");
+
       const namespace = this.emitter.getContext().namespace;
+      const className = ensureCSharpIdentifier(this.emitter.getProgram(), model, name);
       const doc = getDoc(this.emitter.getProgram(), model);
       const attributes = getModelAttributes(this.emitter.getProgram(), model, className);
+      const exceptionConstructor = isErrorType
+        ? this.getModelExceptionConstructor(this.emitter.getProgram(), model, className)
+        : "";
+
       this.#metadateMap.set(model, new CSharpType({ name: className, namespace: namespace }));
       const decl = this.emitter.result.declaration(
         className,
         code`${this.#generatedFileHeader}
 
       ${this.#emitUsings()}
-      
       namespace ${namespace} {
 
-      ${doc ? `${formatComment(doc)}\n` : ""}${`${attributes.map((attribute) => attribute.getApplicationString(this.emitter.getContext().scope)).join("\n")}${attributes?.length > 0 ? "\n" : ""}`}public partial class ${className} ${
-        model.baseModel ? code`: ${this.emitter.emitTypeReference(model.baseModel)}` : ""
-      } {
-      ${this.emitter.emitModelProperties(model)}
+      ${doc ? `${formatComment(doc)}\n` : ""}${`${attributes.map((attribute) => attribute.getApplicationString(this.emitter.getContext().scope)).join("\n")}${attributes?.length > 0 ? "\n" : ""}`}public partial class ${className} ${baseClass} {
+      ${exceptionConstructor ? `${exceptionConstructor}\n` : ""}${this.emitter.emitModelProperties(model)}
     }
    } `,
       );
 
       return decl;
     }
 
+    getModelExceptionConstructor(program: Program, model: Model, className: string): string {
```

Comment:
I really like string | undefined as a type, rather than using `""` to mean the same thing.